pkg/operator/ceph/csi/predicate.go

```diff
@@ -48,12 +50,24 @@ func predicateController(ctx context.Context, c client.Client) predicate.Funcs {
 				if controller.DuplicateCephClusters(ctx, c, e.Object, false) {
 					return false
 				}
+
+				// We still have users that don't use the ConfigMap to configure the
+				// operator/csi and still rely on environment variables from the operator pod's
+				// spec. So we still want to reconcile the controller if the ConfigMap cannot be found.
+				err := c.Get(ctx, types.NamespacedName{Name: opcontroller.OperatorSettingConfigMapName, Namespace: opNamespace}, &v1.ConfigMap{})
+				if err != nil && kerrors.IsNotFound(err) {
+					logger.Debugf("could not find operator configuration ConfigMap, will reconcile the csi controller")
+					return true
```

Comment:
So this means clusters deployed with helm will always reconcile the csi driver every time there is a cephcluster event, but non-helm clusters will only reconcile the first time or when the configmap is updated?

Reply:
Really we just want to ignore the reconcile if the cluster CR status was updated, right? Can we detect that case, but always reconcile the csi driver if the cluster CR spec was updated?

Reply:
The block is just the CREATE sequence, which is called either during the controller initialization (op restart) or when a CephCluster/CM is being created. The update is handled differently.

Reply:
Ok, to summarize our discussion, this CREATE sequence will allow the csi driver to be reconciled at least once after every operator restart.